### PR TITLE
Fix memory leak in python client shared memory

### PR DESCRIPTION
**Pull request by CoderHam**
- must munmap before shm_unlink

### CODE REVIEW

File: src/clients/python/library/tritonclient/utils/shared_memory/__init__.py [hunk 1]
```diff
@@ -252,7 +252,8 @@ def __init__(self, err):
             -2: "unable to get shared memory descriptor",
             -3: "unable to initialize the size",
             -4: "unable to read/mmap the shared memory region",
-            -5: "unable to unlink the shared memory region"
+            -5: "unable to munmap the shared memory region",
```
Comment (GuanLuo) [inline src/clients/python/library/tritonclient/utils/shared_memory/__init__.py:255]: Don't change the defined error, just append the new type of error

Comment (CoderHam) [reply src/clients/python/library/tritonclient/utils/shared_memory/__init__.py:255]: Fixed :)

Comment (GuanLuo) [reply src/clients/python/library/tritonclient/utils/shared_memory/__init__.py:255]: Did you push your fix to remote?

Comment (CoderHam) [reply src/clients/python/library/tritonclient/utils/shared_memory/__init__.py]: Changes had not gotten pushed. Just did it.

File: src/clients/python/library/tritonclient/utils/shared_memory/shared_memory.cc [hunk 2]
```diff
@@ -130,12 +130,21 @@ GetSharedMemoryHandleInfo(
...
+  SharedMemoryHandle* handle =
+      reinterpret_cast<SharedMemoryHandle*>(shm_handle);
+  char* shm_addr = reinterpret_cast<char*>(handle->base_addr_);
+  size_t byte_size = handle->byte_size_;
+  int status = munmap(shm_addr, byte_size);
+  if (status == -1) {
+    return -6;
+  }
+
+  std::string shm_key = handle->shm_key_.c_str();
```
Comment (tanmayv25) [inline src/clients/python/library/tritonclient/utils/shared_memory/shared_memory.cc:142]: You can directly pass the `handle->shm_key_.c_str()` in the shm_unlink function below. No need to create an additional string.

Comment (CoderHam) [reply src/clients/python/library/tritonclient/utils/shared_memory/shared_memory.cc]: Done

**Review by tanmayv25 — APPROVED** [review]

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**